packages/tse-backend/src/common/utils/trading-strategy.utils.ts

```diff
@@ -62,3 +68,122 @@ export const isExchangeSupported = (
 ) => {
   return supported.includes(exchangeName);
 };
+
+type PriceSourceFunction = (orderBook: any, ticker: any) => number;
+
+const priceSourceFunctions: Record<PriceSourceType, PriceSourceFunction> = {
+  [PriceSourceType.MID_PRICE]: (orderBook) =>
+    (orderBook.bids[0][0] + orderBook.asks[0][0]) / 2,
+  [PriceSourceType.BEST_ASK]: (orderBook) => orderBook.asks[0][0],
+  [PriceSourceType.BEST_BID]: (orderBook) => orderBook.bids[0][0],
+  [PriceSourceType.LAST_PRICE]: (_, ticker) => ticker.last,
+};
+
+export const getPriceSource = async (
+  exchange: any,
+  pair: string,
+  priceSourceType: PriceSourceType,
+) => {
+  const orderBook = await exchange.fetchOrderBook(pair);
+  const ticker =
+    priceSourceType === PriceSourceType.LAST_PRICE
+      ? await exchange.fetchTicker(pair)
+      : null;
```

Comment:
Why is ticker `null` if priceSourceType != `PriceSourceType.LAST_PRICE`

Reply:
The ticker is set to null for price sources other than LAST_PRICE because only the order book is needed for those, so there's no need to fetch the ticker:
```
  [PriceSourceType.BEST_BID]: (orderBook) => orderBook.bids[0][0],
  [PriceSourceType.LAST_PRICE]: (_, ticker) => ticker.last,
```